patroni/dcs/etcd.py

```diff
@@ -297,14 +297,20 @@ def _load_machines_cache(self):
 
 
 def catch_etcd_errors(func):
-    def wrapper(*args, **kwargs):
+    def wrapper(self, *args, **kwargs):
         try:
-            return func(*args, **kwargs) is not None
+            return func(self, *args, **kwargs) is not None
         except (RetryFailedError, etcd.EtcdException):
             return False
-        except:
-            logger.exception("")
+        except Exception as e:
+            if not self._has_failed:
+                logger.exception("")
+            else:
+                logger.error(e)
+            self._has_failed = True
             raise EtcdError("unexpected error")
+        else:
+            self._has_failed = False
```

Comment:
Yeah, try-else stubbornly keeps working in differently from what I need of it. Fixed this one.